WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/ApllicationLogDetailView.swift

```diff
@@ -0,0 +1,103 @@
+import SwiftUI
+
+/// Hosting controller that wraps an `ApplicationLogDetailView`
+///
+final class ApplicationLogDetailViewController: UIHostingController<ApplicationLogDetailView> {
+    init(with contents: String, for date: String) {
+        let viewModel = ApplicationLogViewModel(logText: contents)
+        super.init(rootView: ApplicationLogDetailView(viewModel: viewModel))
+        self.title = date
+        viewModel.present = { [weak self] vc in
+            self?.present(vc, animated: true, completion: nil)
+        }
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+struct ApplicationLogDetailView: View {
+    @ObservedObject var viewModel: ApplicationLogViewModel
+
+    private let dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateStyle = .medium
+        formatter.timeStyle = .medium
+        return formatter
+    }()
+
+    var body: some View {
+        ScrollViewReader { scrollProxy in
+            List(viewModel.lines) { line in
+                VStack(alignment: .leading, spacing: 6) {
+                    line.date
+                        .flatMap(dateFormatter.string(for:))
+                        .map {
+                            Text($0)
+                                .footnoteStyle()
+
+                        }
+                    Text(line.text)
+                        .bodyStyle()
+                }
+                .storeAppearance(on: $viewModel.lastCellIsVisible, if: viewModel.isLastLine(line))
+            }
+            .overlay(
+                scrollToBottomButton {
+                    withAnimation {
+                        scrollProxy.scrollTo(viewModel.lastLineID)
+                    }
+                },
+                alignment: .bottomTrailing)
+        }
+        .navigationBarItems(
+            trailing: Button(
+                action: {
+                    viewModel.showShareActivity()
+                }, label: {
+                    Image(systemName: "square.and.arrow.up")
+            }))
+    }
+
+    func scrollToBottomButton(_ action: @escaping () -> Void) -> some View {
+        Group {
+            if viewModel.buttonVisible {
```

Comment:
nit: a `guard let` could make this easier to read with one less level of branching!

Reply:
TIL!